Cookie hardening

jonasfj opened the pull request.

isoos commented on app/lib/frontend/handlers/account.dart:
```diff
@@ -42,12 +42,25 @@ Future<shelf.Response> updateSessionHandler(shelf.Request request,
+      // top-level navigation to the site, as recommended in:
+      // https://tools.ietf.org/html/draft-ietf-httpbis-rfc6265bis-02#section-8.8.2
+      'SameSite=Lax',
+      'Secure', // Only allow this cookie to be sent when making HTTPS requests.
```
can we have a conditional for localhost?
e.g. 
```
if (request.requestedUri.scheme != 'http') 'Secure',
```
(or even `if (!request.requestedUri.toString().startsWith('http://localhost:8080/') 'Secure',`

jonasfj replied:
I think we should somehow put into the `configuration` whether the process was started with `--local-dev` and only in that case should we do something special.

People might be hitting the service with `Host: localhost` in production, you never know. Certainly there is no good reason to allow things like that in production. Let's be defensive :)

isoos replied:
++

jonasfj commented:
@isoos, please take a look, I did a lot of changes since last review.. Now it should work on staging and on localhost (provided you're using the right host).

I also added more hosts to the oauth2 flow for our staging site, so we can use the version specific host name.

jonasfj commented:
This currently deployed to `https://jonasfj-dot-dartlang-pub-dev.appspot.com`

isoos commented on app/lib/account/session_cookie.dart:
```diff
@@ -0,0 +1,91 @@
+
+  // The cookieString is separated by '; ', and contains 'name=value'
+  // See: https://tools.ietf.org/html/rfc6265#section-5.4
+  final cookie = cookieString.split('; ').firstWhere(
```
Nit: We have `parseCookieHeader` in `shared/utils.dart`.

jonasfj replied:
Okay, did a few tweaks and adopted this..
